lib/internal/process.js

```diff
@@ -283,6 +283,43 @@ function setupUncaughtExceptionCapture(exceptionHandlerState) {
   };
 }
 
+function setupCompareVersion() {
+  const {
+    majorVersion,
+    minorVersion,
+    patchVersion,
+    prereleaseTag,
+  } = process.release;
+
+  process.release.compareVersion =
+    (major, minor, patch, prerelease = false) => {
+      if (major > majorVersion)
+        return -1;
+      if (major < majorVersion)
+        return 1;
+
+      if (minor > minorVersion)
+        return -1;
+      if (minor < minorVersion)
+        return 1;
+
+      if (prereleaseTag) {
+        if (prereleaseTag === prerelease && patch === patchVersion)
```

Comment:
I would write this as:

```js
if (prereleaseTag) {
  if (patch === patchVersion)
    return (prereleaseTag === prerelease ? 0 : 1);
}
```

Reply:
Actually scratch that. How about this for the entire bottom part?:

```js
if (patch > patchVersion)
  return -1;
if (patch < patchVersion)
  return 1;
if (prereleaseTag)
  return (prereleaseTag === prerelease ? 0 : 1);
if (prerelease)
  return -1;

return 0;
```

Reply:
Or shorten the last two `return`s even further to just `return (prerelease ? -1 : 0);`